h/static/scripts/group-forms/components/CreateGroupForm.tsx

```diff
@@ -140,6 +143,7 @@ export default function CreateGroupForm() {
       )) as CreateGroupAPIResponse;
     } catch (apiError) {
       setErrorMessage(apiError.message);
+      setSaving(false);
```

Comment:
If you move this into a `finally` block, it will hide the loading spinner even if successful. This would be one way to avoid it being shown if the page is restored from the bfcache for example. When the navigation starts, it is less important to show our own loading state since the browser's native loading hints will be shown.

Reply:
I'm not a fan of the component exiting its loading state when it receives a successful response: it's weird for the spinner to disappear when things are in fact still loading (it looks as if the process has finished and nothing has happened). There is a different spinner being shown in the browser tab but the user may not notice that and even if they do, it still feels weird to me. Also, we don't want the button to be briefly re-enabled before the page navigates: this also looks weird, and may even allow some double clicks.

I've tried another approach which is to exit the loading state if the page is re-shown by the browser's bfcache feature. I think `useEffect()` is the right thing to use here? https://github.com/hypothesis/h/pull/8836/commits/b2ea6aa8711764e43329d4974043aad038e160c9